components/NavBar/index.js

```diff
@@ -1,68 +1,224 @@
-          </Link>
-        </li>
-      </ul>
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetch(USER_DATA_URL, { credentials: 'include' })
+        .then((response) => response.json())
+        .then((responseJson) => {
+          if (responseJson.incompleteUserDetails) {
+            return window.location.replace(
+              'https://my.realdevsquad.com/signup'
+            );
+          }
+          responseJson.statusCode === 401
+            ? setIsLoggedIn(false)
+            : setIsLoggedIn(true);
+
+          setUserData({
+            userName: responseJson.username,
+            firstName: responseJson.first_name,
+          });
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+      setMountedComponent(true);
+    };
+
+    fetchData();
+  }, []);
 
-      <div
-        className={styles.profilePic}
-        ref={navbarRef}
-        onClick={() => setToggle(!toggle)}
-      >
-        <img src={photo} alt="Profile Image" height="70" width="70" />
+  return (
+    <div className={styles.wrapper}>
+      <nav className={styles.navBar}>
+        <div
+          className={styles.hamburger}
+          ref={navbarRef}
+          onClick={() => setToggle(!toggle)}
+        >
+          <span className={styles.bar}></span>
+          <span className={styles.bar}></span>
+          <span className={styles.bar}></span>
+        </div>
         <div
           className={
-            toggle ? styles.dropdownContent : styles.dropdownContentHide
+            mountedComponent
+              ? `${styles.navBarLogin}`
+              : `${styles.navBarLogin} d-none`
           }
         >
-          <a href="#">Link 1</a>
-          <a href="#">Link 2</a>
-          <a href="#">Link 3</a>
-          <a href="#">Link 4</a>
-          <a href="#">Link 5</a>
+          <Link href={LOGIN_URL}>
+            <a
+              className={
+                isLoggedIn ? `${styles.btnLogin} d-none` : `${styles.btnLogin}`
+              }
+            >
+              <button className={styles.btnLoginText}>
+                Sign In
+                <img
+                  className={styles.githubLogo}
+                  src={GITHUB_LOGO}
+                  alt="GitHub Icon"
+                  height="15px"
+                  width="15px"
+                />
+              </button>
+            </a>
+          </Link>
+          <div
+            className={
+              isLoggedIn ? `${styles.userGreet}` : `${styles.userGreet} d-none`
+            }
+          >
+            <div className={styles.userGreetMsg}>
+              {isLoggedIn ? `Hello, ${userData.firstName}!` : `Hello, User!`}
+            </div>
+            <img
+              className={styles.userProfilePic}
+              src={
+                isLoggedIn
+                  ? `https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members/${userData.userName}/img.png`
+                  : ``
```

Comment:
Let's have a dummy picture instead. What do you think?

Reply:
Yes, it is a good idea to add a dummy picture